mm2src/crypto/src/hw_ctx.rs

```diff
@@ -1,57 +1,96 @@
 use crate::crypto_ctx::{MM2_INTERNAL_DERIVATION_PATH, MM2_INTERNAL_ECDSA_CURVE};
-use crate::hw_client::{HwClient, HwError, HwResult};
-use crate::trezor::TrezorClient;
+use crate::hw_client::{HwClient, HwError, HwProcessingError, TrezorConnectProcessor};
+use crate::trezor::TrezorSession;
 use crate::HwWalletType;
 use bip32::ExtendedPublicKey;
+use common::log::warn;
 use common::mm_error::prelude::*;
 use hw_common::primitives::DerivationPath;
 use keys::Public as PublicKey;
+use parking_lot::Mutex as PaMutex;
 use primitives::hash::H264;
 use std::str::FromStr;
-use trezor::response_channel::TrezorResponseReceiver;
-use trezor::TrezorCoin;
+use trezor::client::TrezorClient;
+use trezor::{ProcessTrezorResponse, TrezorCoin, TrezorRequestProcessor};
 
 pub(crate) const MM2_TREZOR_INTERNAL_COIN: TrezorCoin = TrezorCoin::Komodo;
 
 pub struct HardwareWalletCtx {
     /// The pubkey derived from `MM2_INTERNAL_DERIVATION_PATH`.
     pub(crate) mm2_internal_pubkey: H264,
     pub(crate) hw_wallet_type: HwWalletType,
+    /// Please avoid locking multiple mutexes.
+    /// The mutex hasn't be locked while the wallet is used
+    /// because every variant of the Hardware Wallet client uses an internal mutex to operate with the device.
+    /// Clone the `Option<HwClient>` instance instead.
+    pub(crate) hw_wallet: PaMutex<Option<HwClient>>,
```

Comment:
Do we assume that we can reinitialize HwClient while MM2 is running?

It looks like I forgot to reinitialize the `hw_wallet` field if we reconnect to the device...

> Also, is it worth using Arc instead of cloning the entire client?

https://github.com/KomodoPlatform/atomicDEX-API/blob/6882b382d11341970dff7bc979d4f5bab1a9d620/mm2src/crypto/src/hw_client.rs#L110-L113

Where `TrezorClient`

https://github.com/KomodoPlatform/atomicDEX-API/blob/6882b382d11341970dff7bc979d4f5bab1a9d620/mm2src/trezor/src/client.rs#L16-L19

It's designed the same as `UtxoRpcClientEnum` and `ElectrumClient`/`NativeClient`, so there is no cloning of the entire client.
But since it's not clear, I can wrap `HwClient` to `Arc`.

Reply:
> It looks like I forgot to reinitialize the hw_wallet field if we reconnect to the device...

Would like to fix it in this PR or we can move to the next iteration? :slightly_smiling_face: 

> It's designed the same as UtxoRpcClientEnum and ElectrumClient/NativeClient, so there is no cloning of the entire client.
But since it's not clear, I can wrap HwClient to Arc.

It's NP then, please feel free to leave it as is.

Reply:
I've prepared the fix already, just wanted to push it with another your change request if it's needed :)
Ok, thanks

Reply:
Done